pipelines/qc/Snakefile

```diff
@@ -1,32 +1,69 @@
 import os
 import numpy as np
 import pandas as pd
-configfile: "config.yml"
 
-samples = pd.read_csv(config["manifest"], index_col=0)
-script_dir = config["scripts_loc"] 
+sampleset = os.path.split(os.getcwd())[1]
+if os.path.isfile("configuration.yml"):
+  os.remove("configuration.yml")
 
-rule all:
-  input: 
-    contam=expand("output/{sample}.contamination.csv", sample=samples.index.tolist()),
-    alignm=expand("output/{sample}.callstats.csv", sample=samples.index.tolist())
+# read_fofn
+fofn_path = os.path.join(config["vector_ops"], "tracking", sampleset, "agam.fofn.tsv")
+fofn = pd.read_table(fofn_path)
+samples = fofn['sample'].unique()
+
+# set script dir
+script_dir = "../../scripts"
+
+# create a table of paths
+path_df = pd.DataFrame(index=samples)
+path_df["path"] = [os.path.join(config["input_location"], x + ".zarr.zip") for x in samples]
+
+# HACK NOT IN PRODUCTION>>>
+path_df["exists"] = path_df.path.map(os.path.isfile)
+samples = path_df.loc[path_df.exists].index.tolist()
+print(len(samples))
+
+
+rule qc_report:
+  input:
+    manifest="manifest",
+    csv=expand("{sample}.callstats.csv", sample=samples),
+    npy=expand("{sample}.covhist.npz", sample=samples),
+    contam=expand("{sample}.contamination.csv", sample=samples),
+    cfg="configuration.yml"
```

Comment:
Why we cannot specify original config here?

Reply:
This forces snakemake to make a copy of `config.yml` to provide a record of the file in the output dir.

Reply:
smart, agreed